src/lib/externalAttachment/index.ts

```diff
@@ -0,0 +1,92 @@
+import { singleton } from 'tsyringe'
+import { Logger } from 'pino'
+import { logger } from '../logger.js'
+import Identity from '../identity.js'
+import env from '../../env.js'
+import { AttachmentRow } from '../db/types.js'
+
+@singleton()
+export class ExternalAttachmentService {
+  private log: Logger
+
+  constructor(private identity: Identity) {
+    this.log = logger.child({ service: 'ExternalAttachment' })
+  }
+
+  async getOidcConfig(oidcConfigUrl: string) {
+    const response = await fetch(oidcConfigUrl)
+    if (!response.ok) {
+      throw new Error('Failed to fetch OIDC configuration')
+    }
+    return response.json()
+  }
+
+  async getAccessToken(tokenUrl: string, clientId: string, clientSecret: string) {
+    const response = await fetch(tokenUrl, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+      body: new URLSearchParams({
+        grant_type: 'client_credentials',
+        client_id: clientId,
+        client_secret: clientSecret,
+      }),
+    })
+
+    if (!response.ok) {
+      throw new Error('Failed to obtain access token')
+    }
+
+    const data = await response.json()
+    return data.access_token
+  }
+
+  async fetchAttachment(attachmentUrl: string, accessToken: string) {
+    try {
+      const response = await fetch(attachmentUrl, {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+          Accept: 'application/octet-stream',
+        },
+      })
+      const headers = response.headers
+
+      const filename = headers.get('content-disposition')?.split('filename=')[1]?.replace(/['"]/g, '')
```

Comment:
I think we should aim to parse this more carefully. Have a look at https://www.npmjs.com/package/content-disposition and https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Disposition#syntax